docs/source/user_guide/best_practices.rst

```diff
@@ -0,0 +1,167 @@
+
+.. code-block:: python
+
+   >>> import databricks.koalas as ks
+   >>> kdf = ks.DataFrame({'id': range(10)}).sort_values(by="id")
+   >>> kdf.to_spark().explain()
+   == Physical Plan ==
+   *(2) Sort [id#76L ASC NULLS LAST], true, 0
+   +- Exchange rangepartitioning(id#76L ASC NULLS LAST, 200), true, [id=#130]
+      +- *(1) Project [id#76L]
+         +- *(1) Scan ExistingRDD[__index_level_0__#75L,id#76L]
+
+As you can see it requires ``Exchange`` which requires a shuffle and it is likely expensive.
+
+
+Avoid computation on single partition
+-------------------------------------
+
+Another common case is the computation on single partition. Currently some APIs such as
+`DataFrame.rank <https://koalas.readthedocs.io/en/latest/reference/api/databricks.koalas.DataFrame.rank.html>`_
+uses PySpark’s Window without specifying partition specification. This leads to move all data into single
+partition in single machine and could cause serious performance degradation.
+Such APIs shoild be avoided very large dataset.
+
+.. code-block:: python
+
+   >>> import databricks.koalas as ks
+   >>> kdf = ks.DataFrame({'id': range(10)}).sort_values(by="id")
+   >>> kdf.to_spark().explain()
+   >>> kdf.rank().to_spark().explain()
+   == Physical Plan ==
+   *(4) Project [id#547]
+   +- Window [avg(cast(_w0#549 as bigint)) windowspecdefinition(id#521L, specifiedwindowframe(RowFrame, unboundedpreceding$(), unboundedfollowing$())) AS id#547], [id#521L]
+      +- *(3) Project [_w0#549, id#521L]
+         +- Window [row_number() windowspecdefinition(id#521L ASC NULLS FIRST, specifiedwindowframe(RowFrame, unboundedpreceding$(), currentrow$())) AS _w0#549], [id#521L ASC NULLS FIRST]
+            +- *(2) Sort [id#521L ASC NULLS FIRST], false, 0
+               +- Exchange SinglePartition, true, [id=#701]
+                  +- *(1) Project [id#521L]
+                     +- *(1) Scan ExistingRDD[__index_level_0__#520L,id#521L]
+
+Instead, use 
+`GroupBy.rank <https://koalas.readthedocs.io/en/latest/reference/api/databricks.koalas.groupby.GroupBy.rank.html>`_
+as it is less expensive because data can be distributed and computed on each group.
+
+
+Avoid reserved column names
+---------------------------
+
+Columns with leading ``__`` and trailing ``__`` are reserved in Koalas. To handle internal behaviors for, such as, index,
+Koalas uses some internal columns. Therefore it is discouraged to use such column names and not guaranteed to work.
+
+
+Do not use duplicated column names
+----------------------------------
+
+It is disallowed to use duplicated column names because Spark SQL does not allow this in general. Koalas inherits
+this behavior. For instance, see below:
+
+.. code-block:: python
+
+   >>> import databricks.koalas as ks
+   >>> kdf = ks.DataFrame({'a': [1, 2], 'b':[3, 4]})
+   >>> kdf.columns = ["a", "a"]
+   ...
+   Reference 'a' is ambiguous, could be: a, a.;
+
+
+Use ``distributed`` or ``distributed-sequence`` default index
```

Comment:
We should add one more section around this to encourage to use `index_col` when converting Spark DataFrame or using read functions.

Reply:
We still need to consider about default index since there are some functions to loose index and attach default index like `reset_index`, though.

Reply:
Yeah, let me add that point too.